pyccel/ast/basic.py

```diff
@@ -10,29 +10,25 @@
 - PyccelAstNode which describes each PyccelAstNode
 """
 import ast
-from sympy.core.basic import Basic as sp_Basic
 
 __all__ = ('Basic', 'PyccelAstNode')
 
 dict_keys   = type({}.keys())
 dict_values = type({}.values())
 iterable_types = (list, tuple, dict_keys, dict_values)
+iterable = lambda x : isinstance(x, iterable_types)
```

Comment:
where do you use this ?

Reply:
It replaces [this](https://github.com/pyccel/pyccel/blob/master/pyccel/ast/core.py#L17)

Reply:
I am now using it more systematically. f943dbd

Reply:
In order to know if an object is iterable, shouldn't we check it with
```python
hasattr(x, '__iter__')
```
?

Reply:
I don't think so. With that definition we would also include things like `TupleVariable`. In `Basic` this is definitely not what is wanted. I don't think it is what is wanted in core.py either as it is usually used to confirm that the object can be stored in a CodeBlock

Reply:
Maybe there is a better name for this? tuple_like?

Reply:
It's fine, thanks